app/models/course.rb

```diff
@@ -393,6 +394,14 @@ def has_vacancies?
     end
   end
 
+  def has_multiple_running_sites_or_study_modes?
+    running_site_statuses.length > 1 || full_time_or_part_time?
+  end
+
+  def running_site_statuses
+    site_statuses.select(&:running?)
```

Comment:
Has this made it really expensive as it outside or active record land?

Reply:
This was copy pasta'd from old publish code and I was meant to refactor after adding coverage - good spot

Reply:
fixed